ingestion/src/metadata/ingestion/source/databricks.py

```diff
@@ -19,6 +26,69 @@
 from metadata.ingestion.source.sql_source_common import SQLConnectionConfig
 
 
+class STRUCT(String):
+    """The SQL STRUCT type."""
+
+    __visit_name__ = "STRUCT"
```

Comment:
Could you please add some comments on these classes? Just to understand why are we following this structure and how it is applied

Reply:
The pyhive lib maps the struct, array and map types to [string](https://github.com/dropbox/PyHive/blob/3547bd6cccf963a033928b73c5ed498684335c39/pyhive/sqlalchemy_hive.py#L122) sqltype of sqlalchamy package, which does not support the struct, array and map types by default. So I've create the custom class for these datatypes referring to [Varchar](https://github.com/sqlalchemy/sqlalchemy/blob/d7caa3c0bef64c710e45ed646756da65c4241bdd/lib/sqlalchemy/sql/sqltypes.py#L3149) datatype. I also overridden the [get_columns](https://github.com/crflynn/sqlalchemy-databricks/blob/b2d2215257a1a2d9e4a24d9dc9e348f7cd9bcac6/sqlalchemy_databricks/_dialect.py#L39) function to add the raw_data_type so that our framework handles these datatypes

Reply:
aaah I see, thanks for the explanation. Also I see that the function was extracted from the dialect itself. Could you please add this as a comment to the module or the classes in the code?

Thanks 🙏

Reply:
sure